### PR TITLE
Restructure ImageBitmap tests

### PR DESCRIPTION
**Pull request by xidachen**
At this moment, there are quite some repetitive code resides in the ImageBitmap testing scripts. This change extract those common functions into a "tex-image-and-sub-image-2d-with-image-bitmap-utils.js", and simplify the testing scripts a lot.

Locally, the testing results are the same as what it is without this change.

Please review. @kenrussell @zhenyao @toji

### CODE REVIEW

**Comment by zhenyao** [comment]
It didn't display all the changes due to too many files.

Can you upload the python file and the manually written js file for review first?  Then all the auto-generated changes in a second patch.

**Comment by xidachen** [comment]
@zhenyao Thank you for the comment. At this moment, only those manual js testing scripts and the python script are changed. It should be more clear now.

File: sdk/tests/js/tests/tex-image-and-sub-image-2d-with-image-bitmap-utils.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        { sub: true, bitmap: bitmaps.flipYUnpremul, flipY: true, premultiply: false },
+    ];
+
+    for (var i in cases) {
+        runOneIterationImageBitmapTest(cases[i].sub, bindingTarget, program, cases[i].bitmap,
+            cases[i].flipY, cases[i].premultiply, optionsVal, internalFormat, pixelFormat, pixelType, gl, tiu, wtu);
+    }
+}
+
+function runImageBitmapTest(bitmaps, alphaVal, internalFormat, pixelFormat, pixelType, gl, tiu, wtu)
```
Comment (kenrussell) [inline sdk/tests/js/tests/tex-image-and-sub-image-2d-with-image-bitmap-utils.js]: Could you rename this to "runImageBitmapTestImpl" or "runImageBitmapTestInternal" or something similar to indicate it's non-public?

File: sdk/tests/js/tests/tex-image-and-sub-image-2d-with-image-bitmap-utils.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    if (bitmaps.noFlipYPremul.width == bitmaps.noFlipYPremul.height) {
+        program = tiu.setupTexturedQuadWithCubeMap(gl, internalFormat);
+        runTestOnBindingTargetImageBitmap(gl.TEXTURE_CUBE_MAP, program, bitmaps, optionsVal,
+            internalFormat, pixelFormat, pixelType, gl, tiu, wtu);
+    }
+
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "should be no errors");
+}
+
+function generateImageBitmap(source, alphaVal, internalFormat, pixelFormat, pixelType, gl, tiu, wtu)
```
Comment (kenrussell) [inline sdk/tests/js/tests/tex-image-and-sub-image-2d-with-image-bitmap-utils.js]: Could you please rename this to something like "runImageBitmapTest"? It has multiple callers, and it's confusing that they call both "createImageBitmap" and "generateImageBitmap".

**Comment by kenrussell** [comment]
I'm sorry I dropped this review on the floor. Xida, could you please address the couple of comments above?

**Comment by xidachen** [comment]
Hi Ken,
Thanks for reviewing it. I have addressed the comments.

**Comment by xidachen** [comment]
Also, I should have mentioned that once this is approved, I will run the auto-generate python script to update the html files before merge this.

**Comment by kenrussell** [comment]
Very nice. LGTM. Please run the generator script and post here when done; will merge it then.

**Comment by xidachen** [comment]
I have run the generate script and all html files are updated. Please review.

**Comment by kenrussell** [comment]
Checked a couple of the autogenerated files and they look fine. Thank you for doing this cleanup. Merging now.

**Pull request merged**